test/rule-matches/color-contrast-matches.js

```diff
@@ -22,216 +24,326 @@ describe('color-contrast-matches', function () {
+		assert.isFalse(rule.matches(target, axe.utils.getNodeFromTree(tree[0], target)));
 	});
-});
\ No newline at end of file
+
+	if (shadowSupport) {
+		it('should match a descendant of an element across a shadow boundary', function () {
+			fixture.innerHTML = '<div id="parent" style="background-color: #000;">' +
+				'</div>';
+
+			var shadowRoot = document.getElementById('parent').attachShadow({ mode: 'open' });
+			shadowRoot.innerHTML = '<div id="shadowTarget" style="color: #333">Text</div>';
+
+			var shadowTarget = fixture.firstChild.shadowRoot.querySelector('#shadowTarget');
+			var tree = axe._tree = axe.utils.getFlattenedTree(fixture);
+			assert.isTrue(rule.matches(shadowTarget, axe.utils.getNodeFromTree(tree[0], shadowTarget)));
+		});
+
+		it('should look at the correct root node when looking up an explicit label and disabled input', function () {
+			fixture.innerHTML = '<div id="parent">'+
+				'<input id="input">' +
+			'</div>';
+
+			var shadowRoot = document.getElementById('parent').attachShadow({ mode: 'open' });
+			shadowRoot.innerHTML = '<div id="shadowParent">' +
+					'<label for="input" id="shadowLabel">Label</label>' +
+					'<input id="input" disabled>' +
+				'</div>';
+
+			var shadowLabel = fixture.firstChild.shadowRoot.querySelector('#shadowLabel');
+			var tree = axe._tree = axe.utils.getFlattenedTree(fixture);
+			assert.isFalse(rule.matches(shadowLabel, axe.utils.getNodeFromTree(tree[0], shadowLabel)));
+		});
+
+		it('should look at the correct root node when looking up implicit label and disabled input', function () {
+			fixture.innerHTML = '<div id="parent">'+
+				'<input id="input">' +
+			'</div>';
+
+			var shadowRoot = document.getElementById('parent').attachShadow({ mode: 'open' });
+			shadowRoot.innerHTML = '<div id="shadowParent">' +
+					'<label for="input" id="shadowLabel">Label' +
+					'<input id="input" disabled></label>' +
+				'</div>';
+
+			var shadowLabel = fixture.firstChild.shadowRoot.querySelector('#shadowLabel');
+			var tree = axe._tree = axe.utils.getFlattenedTree(fixture);
+			assert.isFalse(rule.matches(shadowLabel, axe.utils.getNodeFromTree(tree[0], shadowLabel)));
+		});
+
+		it('should look at the correct root node for a disabled control\'s label associated w/ aria-labelledby', function () {
+			fixture.innerHTML = '<div id="parent">'+
+				'<input id="input">' +
+			'</div>';
+
+			var shadowRoot = document.getElementById('parent').attachShadow({ mode: 'open' });
+			shadowRoot.innerHTML = '<div id="shadowParent">' +
+					'<label id="shadowLabel">Label</label>' +
+					'<input aria-labelledby="shadowLabel" disabled>' +
+				'</div>';
+
+			var shadowLabel = fixture.firstChild.shadowRoot.querySelector('#shadowLabel');
+			var tree = axe._tree = axe.utils.getFlattenedTree(fixture);
+			assert.isFalse(rule.matches(shadowLabel, axe.utils.getNodeFromTree(tree[0], shadowLabel)));
+		});
+
+		it('should look at the children of a virtual node for overlap', function () {
```

Comment:
Also that conditional returns true–I thought it was checking to see if there was overlapping content worth evaluating.